apps/examples/testcase/ta_tc/task_manager/itc/itc_task_manager_main.c

```diff
@@ -0,0 +1,282 @@
+}
+
+/**
+ * @testcase         itc_task_manager_clean_info_twice
+ * @brief            Get task_manager_getinfo_with_handle twice
+ * @scenario         Get task_manager_getinfo_with_handle twice
+ * @apicovered       task_manager_getinfo_with_handle
+ * @precondition     NA
+ * @postcondition    NA
+ */
+static void itc_task_manager_clean_info_twice(void)
+{
+	sample_info = (task_info_t *) task_manager_getinfo_with_handle(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_NEQ("task_manager_clean_info", (task_info_t *)&sample_info, NULL);
+
+	task_manager_clean_info(&sample_info);
+	TC_ASSERT_NEQ("task_manager_clean_info", (task_info_t *)&sample_info, NULL);
+
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         itc_task_manager_register_n
+ * @brief            Register task manager with invalid parameter
+ * @scenario         Register task manager with invalid parameter
+ * @apicovered       task_manager_register
+ * @precondition     NA
+ * @postcondition    NA
+ */
+
+static void itc_task_manager_register_n(void)
+{
+	int ret;
+	ret = task_manager_register(NULL, TM_TASK_PERMISSION_DEDICATE,
+	TM_NO_RESPONSE);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME, TM_INVALID_PERMISSION,
+	TM_NO_RESPONSE);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME, TM_TASK_PERMISSION_GROUP,
+	TM_INVALID_TIMEOUT);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME,
+	TM_INVALID_UPPER_RANGE_PERMISSION, TM_INVALID_TIMEOUT);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME, TM_TASK_PERMISSION_GROUP, TM_NO_RESPONSE);
+	TC_ASSERT_NEQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	TC_SUCCESS_RESULT();
+}
+/**
+ * @testcase         itc_task_manager_register_p
+ * @brief            Register task manager with valid parameter
+ * @scenario         Register task manager with valid parameter
+ * @apicovered       task_manager_register
+ * @precondition     NA
+ * @postcondition    NA
+ */
+static void itc_task_manager_register_p(void)
+{
+
+	tm_sample_handle = task_manager_register("invalid",
+	TM_TASK_PERMISSION_DEDICATE, TM_RESPONSE_WAIT_INF);
+
+	TC_ASSERT_EQ("task_manager_register_p", tm_sample_handle,
+			TM_OPERATION_FAIL);
+	tm_sample_handle = task_manager_register(TM_SAMPLE_NAME,
+	TM_TASK_PERMISSION_DEDICATE, TM_RESPONSE_WAIT_INF);
+
+	TC_ASSERT_GEQ("task_manager_register_p", tm_sample_handle, 0);
+
+	TC_SUCCESS_RESULT();
+}
+/**
+ * @testcase         itc_task_manager_start
+ * @brief            Start task manager with different parameter
+ * @scenario         Start task manager with different parameter
+ * @apicovered       task_manager_start
+ * @precondition     NA
+ * @postcondition    NA
+ */
+static void itc_task_manager_start(void)
```

Comment:
how about dividing this TC? : positive and negative

Reply:
yes, we made changes as per your suggestion